components/tasks/TaskTable.tsx

```diff
@@ -0,0 +1,222 @@
+                                        <Td>{el.name}</Td>
+                                        <Td>
+                                            <Button>x</Button>
+                                        </Td>
+                                    </Tr>
+                                );
+                            })}
+                        </Tbody>
+                    </Table>
+                </Box>
+            )}
+            <Flex flexDirection="row-reverse">
+                <Button colorScheme="blue" onClick={onOpen}>
+                    Add Task
+                </Button>
+            </Flex>
+            <Modal isOpen={isOpen} onClose={onClose}>
+                <ModalOverlay />
+                <ModalContent px="0.5rem">
+                    <ModalHeader>Add task to project</ModalHeader>
+                    <ModalCloseButton />
+                    <FormControl isInvalid={state.name.length === 0} isRequired>
+                        <FormLabel>Task Name</FormLabel>
+                        <Input
+                            placeholder="Task Name"
+                            onChange={(e) =>
+                                setState({
+                                    ...state,
+                                    name: e.target.value,
+                                })
+                            }
+                        />
+                        <FormErrorMessage>Task name cannot be zero characters long.</FormErrorMessage>
+                    </FormControl>
+                    <FormControl>
+                        <FormLabel>Description</FormLabel>
+                        <Input
+                            placeholder="Description"
+                            onChange={(e) =>
+                                setState({
+                                    ...state,
+                                    description: e.target.value,
+                                })
+                            }
+                        />
+                    </FormControl>
+                    <FormControl isInvalid={state.startDate.length === 0} isRequired>
+                        <FormLabel>Start Date</FormLabel>
+                        <Input
+                            type="date"
+                            value={state.startDate}
+                            placeholder="Project start date"
+                            onChange={(e) =>
+                                setState({
+                                    ...state,
+                                    startDate: e.target.value,
+                                })
+                            }
+                        ></Input>
+                        <FormErrorMessage>Task start date must not be empty.</FormErrorMessage>
+                    </FormControl>
+                    <FormControl
+                        isInvalid={state.startDate ? (state.endDate ? state.endDate <= state.startDate : false) : false}
```

Comment:
Is this correct that both ternary elses give false or am I just reading this incorrectly? If there's any way to avoid using nested ternary expressions I would avoid them because they are very hard to read

Reply:
It returns true if both dates are defined and end date precedes start date. Otherwise returns false. I don't know how to do it otherwise, as you must check that both are not undefined

Reply:
Which I did in 66133e8. Is that better?

Reply:
I think so. It's much easier to read. 👍🏻 
Writing short helper functions like that is a good habit IMO. On top of that now there's a nice low hanging fruit to improve test coverage, since it would be possible to write a unit test for your new helper function. Though I don't think there's a module for running tests here yet. @robertsiipola should we look into adding `jest` or similar to actually test something in the code?

Reply:
> I think so. It's much easier to read. 👍🏻 Writing short helper functions like that is a good habit IMO. On top of that now there's a nice low hanging fruit to improve test coverage, since it would be possible to write a unit test for your new helper function. Though I don't think there's a module for running tests here yet. @robertsiipola should we look into adding `jest` or similar to actually test something in the code?

Yes `jest` for unit testing helper functions is fine. Apart from that I'd prefer something like Cypress/Selenium/Puppeteer to test the actual frontend. For example Chrome now integrates with Puppeteer, so that you can "record" a user flow from which the browser generates a javascript file for Puppeteer that keeps track of all the user/browser events. https://developer.chrome.com/docs/devtools/recorder/. 

Puppeteer also has a pretty decent integration with jest: https://jestjs.io/docs/puppeteer

Reply:
Date checking moved to utils in fcdfd31